In [1]:
import numpy as np

In [2]:
import pandas as pd

In [3]:
data=pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv',encoding='latin-1')

In [4]:
data

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

In [6]:
data.head

<bound method NDFrame.head of        Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     100

In [7]:
data.tail

<bound method NDFrame.tail of        Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     100

In [9]:
data.shape

(60943, 12)

In [11]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


In [12]:
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [13]:
data.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

# python for meachine learning quiz codes of 11 to 20  questions.

# 11.Question

Answer the following questions based on the African food production dataset provided by the FAO website already provided

What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

Hint:

Perform a groupby sum aggregation on ‘Item’

Options

A. 213450.34 and 35617.22

B. 332319.31 and 125467.80

C. 341432.12 and 331245.67

D. 209460.54 and 269617.53

In [15]:
data.groupby('Item')['Y2014'].sum()

Item
Alcohol, Non-Food        2403.00
Alcoholic Beverages    102410.11
Animal Products         11935.65
Animal fats            209460.54
Apples and products      9499.23
                         ...    
Vegetables, Other      155038.96
Vegetal Products       107145.19
Wheat and products     232670.13
Wine                     4497.36
Yams                   200396.96
Name: Y2014, Length: 119, dtype: float64

In [16]:
data.groupby('Item')['Y2017'].sum()

Item
Alcohol, Non-Food        2348.00
Alcoholic Beverages     95581.06
Animal Products         11547.65
Animal fats            269617.53
Apples and products     10198.90
                         ...    
Vegetables, Other      157752.59
Vegetal Products       107655.20
Wheat and products     240047.62
Wine                     4178.02
Yams                   229174.59
Name: Y2017, Length: 119, dtype: float64

ANS: D. 209460.54 AND 269617.53

# 12 Question
What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

Options

A. 10.60 and 8.49

B. 5.76 and 10.60

C. 135.236 and 1603.404

D. 10.60 and 17.01

E. 8.

In [17]:
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

ANS: C. 135.236 and 1603.404

# 13 Question
What is the total number and percentage of missing data in 2016 to 2 decimal places?

Options

A. 1437 and 2.15 percent

B. 1506 and 2.47

C. 1548 and 2.54 

D. 1506 and 2.47

E. 1535 and 2.52

In [18]:
data.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [19]:
percentage=(1535/60943)*100
percentage

2.5187470259094566

ANS: 'E' 1535 and 2.52

# 14 Question
Which year had the highest correlation with ‘Element Code’? 

Options

A. 2017

B. 2016

C. 2014

D. 2015

In [20]:
data.corr().abs()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000   0.005159      0.000209  0.006164  0.005472   
Item Code      0.005159   1.000000      0.024683  0.021722  0.020857   
Element Code   0.000209   0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

ANS: 'C' 2014

# 15 Question
What year has the highest sum of Import Quantity?

Hint- Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question

Options

A. 2014

B. 2016

C. 2015

D. 2017

In [23]:
data.groupby('Element')['Y2014'].sum()

Element
Domestic supply quantity                  1996716.35
Export Quantity                            150020.64
Fat supply quantity (g/capita/day)          10225.56
Feed                                       216927.89
Food                                      1212332.49
Food supply (kcal/capita/day)              454257.00
Food supply quantity (kg/capita/yr)         49650.63
Import Quantity                            274144.48
Losses                                     153223.00
Other uses (non-food)                       78718.13
Processing                                 282923.00
Production                                1931287.75
Protein supply quantity (g/capita/day)      11836.46
Residuals                                   30149.00
Seed                                        21922.92
Stock Variation                             58749.83
Total Population - Both sexes             1031585.00
Tourist consumption                           416.00
Name: Y2014, dtype: float64

In [24]:
data.groupby('Element')['Y2015'].sum()

Element
Domestic supply quantity                  2021493.55
Export Quantity                            157614.47
Fat supply quantity (g/capita/day)          10235.74
Feed                                       225050.22
Food                                      1232361.10
Food supply (kcal/capita/day)              453383.00
Food supply quantity (kg/capita/yr)         49345.13
Import Quantity                            267018.46
Losses                                     155439.00
Other uses (non-food)                       66254.41
Processing                                 287929.00
Production                                1947019.39
Protein supply quantity (g/capita/day)      11833.95
Residuals                                   30045.00
Seed                                        23976.82
Stock Variation                             34910.99
Total Population - Both sexes             1058081.00
Tourist consumption                           349.00
Name: Y2015, dtype: float64

In [26]:
data.groupby('Element')['Y2016'].sum()

Element
Domestic supply quantity                  2044842.70
Export Quantity                            151920.46
Fat supply quantity (g/capita/day)          10102.77
Feed                                       228958.65
Food                                      1247022.17
Food supply (kcal/capita/day)              451810.00
Food supply quantity (kg/capita/yr)         48985.28
Import Quantity                            286582.78
Losses                                     157787.00
Other uses (non-food)                       69563.68
Processing                                 280631.00
Production                                1943537.15
Protein supply quantity (g/capita/day)      11779.69
Residuals                                   37224.00
Seed                                        23389.20
Stock Variation                             33140.12
Total Population - Both sexes             1085107.00
Tourist consumption                            89.00
Name: Y2016, dtype: float64

In [27]:
data.groupby('Element')['Y2017'].sum()

Element
Domestic supply quantity                  2088198.10
Export Quantity                            182338.80
Fat supply quantity (g/capita/day)          10253.84
Feed                                       223705.68
Food                                      1258888.28
Food supply (kcal/capita/day)              454681.00
Food supply quantity (kg/capita/yr)         48690.04
Import Quantity                            294559.09
Losses                                     160614.00
Other uses (non-food)                       91645.97
Processing                                 292836.00
Production                                2030056.89
Protein supply quantity (g/capita/day)      11842.45
Residuals                                   35500.00
Seed                                        24870.14
Stock Variation                             54316.91
Total Population - Both sexes             1112641.00
Tourist consumption                            91.00
Name: Y2017, dtype: float64

ANS: 'D' 2017

# 16Question 
What is the total number of the sum of Production in 2014?

Hint- Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question

Options

A. 1954537.22

B. 1947019.23

C. 1931287.75

D. 207134.67

E.202118.43

In [28]:
data.groupby('Element')['Y2014'].sum()

Element
Domestic supply quantity                  1996716.35
Export Quantity                            150020.64
Fat supply quantity (g/capita/day)          10225.56
Feed                                       216927.89
Food                                      1212332.49
Food supply (kcal/capita/day)              454257.00
Food supply quantity (kg/capita/yr)         49650.63
Import Quantity                            274144.48
Losses                                     153223.00
Other uses (non-food)                       78718.13
Processing                                 282923.00
Production                                1931287.75
Protein supply quantity (g/capita/day)      11836.46
Residuals                                   30149.00
Seed                                        21922.92
Stock Variation                             58749.83
Total Population - Both sexes             1031585.00
Tourist consumption                           416.00
Name: Y2014, dtype: float64

ANS: 'C' 1931287.75

# 17Question
Which of these elements had the highest sum in 2018?

Hint-  Select columns ‘Y2018’ and ‘Element’, Perform a groupby operation on ‘Element’ on the selected dataframe and answer the  question.

Options

A. Domestic supply quantity

B. Food supply quantity (kg/capita/yr)

C. Food supply (k/cal/capita/day)

D. Import quantity

In [29]:
data.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

ANS: 'A' Domestic supply quantity

# 18Question
Which of these elements had the 3rd lowest sum in 2018?

Hint-  Select columns ‘Y2018’ and ‘Element’, Perform a groupby operation on ‘Element’ on the selected dataframe and answer the  question.  

Options

A. Food supply (kcal/capita/day)

B. Fat supply quantity (g/capita/day)

C. Food supply quantity (kg/capita/yr)

D. Seed

E. Protein supply quantity (g/capita/day)

In [30]:
data.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

ANS: 'E' Protein supply quantity(g/capita/day)

# 19Question
What is the total Import Quantity in Algeria in 2018?

Options

A. 36736.08

B. 37841.91

C. 36059.29

D. 36238.29

In [31]:
data_Algeria=data[data['Area']=='Algeria']

In [33]:
data.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [34]:
data_Algeria.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  90023.42
Export Quantity                             811.54
Fat supply quantity (g/capita/day)          309.09
Feed                                      10668.00
Food                                      63736.33
Food supply (kcal/capita/day)             13288.00
Food supply quantity (kg/capita/yr)        1509.28
Import Quantity                           36238.29
Losses                                     7708.00
Other uses (non-food)                      3970.08
Processing                                 2479.00
Production                                60766.67
Protein supply quantity (g/capita/day)      367.28
Residuals                                   630.00
Seed                                        825.00
Stock Variation                            6172.00
Total Population - Both sexes             42228.00
Name: Y2018, dtype: float64

ANS: 'D'  36238.26

# 20Question
What is the total number of unique countries in the dataset?

Options

A. 50

B. 49

C. 45

D. 39

In [36]:
data['Area'].value_counts().count()

49

ANS: 'B'  49